data_safe_haven/exceptions/__init__.py

```diff
@@ -42,6 +42,14 @@ class DataSafeHavenAzureError(DataSafeHavenCloudError):
     pass
 
 
+class DataSafeHavenAzureAPIError(DataSafeHavenError):
```

Comment:
```suggestion
class DataSafeHavenAzureAPIError(DataSafeHavenAzureError):
```

Reply:
I didn't do that intentionally, as there were a lot of instances of `except DataSafeHavenAzureError` (and some raw `except Exception` :scream:) which would prevent us from catching the API authentication error.

Could be looked at again in #1873